src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -83,55 +86,44 @@ public static object ConvertFromJson(string input, bool returnHashTable, out Err
                     });
 
                 // JObject is a IDictionary
-                var dictionary = obj as JObject;
-                if (dictionary != null)
+                if (obj is JObject dictionary)
                 {
-                    if (returnHashTable)
-                    {
-                        obj = PopulateHashTableFromJDictionary(dictionary, out error);
-                    }
-                    else
-                    {
-                        obj = PopulateFromJDictionary(dictionary, out error);
-                    }
+                    obj = returnHashTable ?
+                              PopulateHashTableFromJDictionary(dictionary, out error) :
+                              PopulateFromJDictionary(dictionary, memberTracker, out error);
                 }
                 else
                 {
                     // JArray is a collection
-                    var list = obj as JArray;
-                    if (list != null)
+                    if (obj is JArray list)
                     {
-                        if (returnHashTable)
-                        {
-                            obj = PopulateHashTableFromJArray(list, out error);
-                        }
-                        else
-                        {
-                            obj = PopulateFromJArray(list, out error);
-                        }
+                        obj = returnHashTable ?
+                                  PopulateHashTableFromJArray(list, out error) :
+                                  PopulateFromJArray(list, out error);
                     }
                 }
             }
             catch (JsonException je)
             {
                 var msg = string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.JsonDeserializationFailed, je.Message);
+
                 // the same as JavaScriptSerializer does
                 throw new ArgumentException(msg, je);
             }
+
             return obj;
         }
 
         // This function is a clone of PopulateFromDictionary using JObject as an input.
-        private static PSObject PopulateFromJDictionary(JObject entries, out ErrorRecord error)
+        private static PSObject PopulateFromJDictionary(JObject entries, DuplicateMemberSet memberTracker, out ErrorRecord error)
         {
             error = null;
-            PSObject result = new PSObject();
+            var result = new PSObject(entries.Count);
             foreach (var entry in entries)
             {
                 if (string.IsNullOrEmpty(entry.Key))
                 {
-                    string errorMsg = string.Format(CultureInfo.InvariantCulture,
-                                                    WebCmdletStrings.EmptyKeyInJsonString);
+                    var errorMsg = string.Format(CultureInfo.InvariantCulture, WebCmdletStrings.EmptyKeyInJsonString);
```

Comment:
minor: `CurrentCulture` should be used when constructing error strings. It's a flaw in the original code.